### ISSUE TITLE
Retirement of OSC Cluster 1

### ISSUE THREAD

**Issue by caldeirav** [os-climate/os_c_data_commons#186]

**Comment by caldeirav** [comment]
This issue is used to track activities towards the retirement of cluster 1 including migration of pipelines.

**Comment by MichaelTiemannOSC** [comment]
For starters, here are the notebooks that reference CL1.  These notebook references should be replaced by tracker issues as they are created:

- [ ] aicoe-osc-demo (demo1, demo2, demo3): https://github.com/os-climate/aicoe-osc-demo/issues/169
- [ ] ./data-platform-demo/notebooks/trino-iceberg-ingestion.ipynb:1
- [ ] ./data-platform-demo/notebooks/itr-ingest-raw-emissions.ipynb:1
- [ ] ./data-platform-demo/notebooks/odsc-west-2021-federations.ipynb:1
- [ ] ./data-platform-demo/notebooks/Sankey.ipynb:2
- [ ] ./data-platform-demo/notebooks/EPA_GHGRP-demos.ipynb:1
- [ ] ./data-platform-demo/notebooks/trino-datavault-test.ipynb:3
- [ ] ./data-platform-demo/notebooks/trino-iceberg-demo.ipynb:1
- [ ] ./data-platform-demo/notebooks/itr-ingest-raw-company-data.ipynb:1
- [ ] ./data-platform-demo/notebooks/itr-ingest-raw.ipynb:2
- [ ] ./data-platform-demo/notebooks/deprecated/trino-access-control-demo.ipynb:2
- [ ] ./data-platform-demo/notebooks/deprecated/test-trino-access.ipynb:1
- [ ] ./data-platform-demo/notebooks/deprecated/trino-api-demo.ipynb:1
- [ ] ./data-platform-demo/notebooks/deprecated/trino-sqlalchemy-demo.ipynb:1
- [ ] ./data-platform-demo/notebooks/deprecated/trino-hive-iceberg-ingest.ipynb:1
- [x] (This will list all available clusters, including CL1 until it is retired ./apps/docs/content/notebooks/onboarding_project.ipynb:1)
- [x] (Of course a CL1 cluster configuration file will reference CL1: ./apps/kfdefs/overlays/osc/osc-cl1/superset/configmaps/superset_config.py:3)


I have created a branch `iceberg` in data-platform-demo that migrates various demos from CL1 to CL2.  Other Pull Requests welcome.

If you are aware of other migration issues or trackers, please feel free to edit above.

**Comment by MichaelTiemannOSC** [comment]
I suggest we delete/archive data-platform-demo/notebooks/itr-* as (1) they are very old, demonstrating usage of a very early version of osc-ingest-tools functionality, (2) the sample data is a public slice of a proprietary dataset that is not presently available in any meaningful way in the Data Commons.  @erikerlandson are you OK with that?

**Comment by redmikhail** [comment]
While majority of the active development should happen on osc-cl2. There is argument to be made that we may want to keep cluster one for new platform/tool testing in scale-down format. As part of it once all pipelines will be migrated we may want to upgrade and sync it with cluster 2

**Comment by eoriorda** [comment]
1. Migrate the relevant or needed workloads to cluster 2
     - Eric to fix ingest script to remove the 4 day latency .  OSC Ingest tools is the .
     - This is the main blocker to prevent us moving to Cluster 2

**Comment by HeatherAck** [comment]
ingest script fixed.  hive catalog is up.  workflow runs; on-boarding new workflows. 
ITR ones left to be moved and AICOE
Mikhail - Openshift Upgrade 4.8 to 4.9, synchronize pipelines and tools that are different from CL2, identify namespaces for Airbus, downsize cluster

**Comment by HeatherAck** [comment]
ingest script fixed.  hive catalog is up.  workflow runs; on-boarding new workflows. 
ITR ones left to be moved and AICOE
Mikhail - Openshift Upgrade 4.8 to 4.9, synchronize pipelines and tools that are different from CL2, identify namespaces for Airbus, downsize cluster

**Comment by eoriorda** [comment]
ITR will be removed by Michael T - as everything is moved to Cluster 2 that is needed. @MichaelTiemannOSC

**Comment by eoriorda** [comment]
Lets wait for Heather to confirm . Seem like work is being done.

**Comment by MichaelTiemannOSC** [comment]
WITNESS and SoS Trades are now fully operational on CL2, so no longer require CL1 resources (per tech check-in meeting today).

**Comment by HeatherAck** [comment]
@redmikhail will create a new set of tasks/issues to perform specific work: (1) review what is left (2) sync CL1 with CL2 with version/config (3) reduce CL1 nodes

**Comment by HeatherAck** [comment]
@redmikhail still pending - to be done by 30-sep

**Comment by HeatherAck** [comment]
@redmikhail still pending - to be done by 30-sep

**Comment by HeatherAck** [comment]
@redmikhail to complete by 10-Oct

**Comment by redmikhail** [comment]
Further steps to re-purpose  cluster cl1 : 

- [ ] Upgrade Openshift version from 4.8 to 4.9 to match with cluster 1 
> - [ ] Check/remove versions of CRD/API that are not compatible with Kubernetes 1.22 (https://access.redhat.com/articles/6955985)
> - [ ] Upgrade cluster to version 4.9 
- [ ] Synchronize management of cluster resources with cluster 2 
> - [ ] Add machinesets to github with proper labeling 
> - [ ] Add taints to GPU enabled nodes 
> - [ ] Add taints to non-GPU jupyterhub nodes 
- [ ] Synchronize setup and versions of OS Climate components: ODH, Trino , Kubeflow, Openmetadata
- [ ] Review capacity of components above and downsize if possible (eg. do we need 3 workers for trino if there is no workload)
- [ ] Document design of the environments (cluster 1 and cluster 2)

**Comment by HeatherAck** [comment]
on hold until 21-nov (post COP27); @redmikhail will reach out to risk mgmt team

**Comment by HeatherAck** [comment]
- the upgrade to open shift to 4.9 is blocked by namespace postgres (physical risk) operator issue
- reduced GPU enabled nodes to 1; namespaces can be scaled down as well
- @redmikhail to let @HeatherAck know the current AWS spend

**Comment by HeatherAck** [comment]
@redmikhail to send AWS spend today

**Comment by HeatherAck** [comment]
AWS spend has been stable. 5 remaining accounts to be removed.  @MichaelTiemann to send @ryanaslett credentials for 2 accounts.  @ryanaslett to install kubecost to understand cost per workload (Heather to create a Jira)

**Comment by HeatherAck** [comment]
@redmikhail to upgrade to new version of Open Shift

**Comment by redmikhail** [comment]
@HeatherAck it has been completed , I have marked tasks in the list above

**Comment by HeatherAck** [comment]
YOU ARE AWESOME!  Thank you!  Thank you!
